chain/stmgr/call.go

```diff
@@ -156,6 +157,10 @@ func (sm *StateManager) CallWithGas(ctx context.Context, msg *types.Message, pri
 	ctx, span := trace.StartSpan(ctx, "statemanager.CallWithGas")
 	defer span.End()
 
+	// Copy the message as we'll be modifying the nonce.
```

Comment:
I don't understand why we need this copy now, and not before?

Reply:
We needed it before as well, technically. In practice, this doesn't matter because there's exactly one caller of this method (gasEstimateGasLimit) and that caller _also_ copies the message.